Source/WebCore/bindings/js/JSExecState.cpp

```diff
@@ -58,4 +60,42 @@ ScriptExecutionContext* executionContext(JSC::JSGlobalObject* globalObject)
     return JSC::jsCast<JSDOMGlobalObject*>(globalObject)->scriptExecutionContext();
 }
 
+JSC::JSValue JSExecState::profiledCall(JSC::JSGlobalObject* lexicalGlobalObject, JSC::ProfilingReason reason, JSC::JSValue functionObject, const JSC::CallData& callData, JSC::JSValue thisValue, const JSC::ArgList& args, NakedPtr<JSC::Exception>& returnedException)
+{
+    JSC::VM& vm = JSC::getVM(lexicalGlobalObject);
+    auto scope = DECLARE_CATCH_SCOPE(vm);
+    JSC::JSValue returnValue;
+    {
+        JSExecState currentState(lexicalGlobalObject);
+        if (Document* document = dynamicDowncast<Document>(jsDynamicCast<JSDOMGlobalObject*>(lexicalGlobalObject)->scriptExecutionContext())) {
+            if (UNLIKELY(document->quirks().needsToRemoveGetFromAccessor())) {
+                SetForScope needsToRemoveGetFromAccessorScope(lexicalGlobalObject->vm().m_needsToRemoveGetFromAccessor, true);
+                returnValue = JSC::profiledCall(lexicalGlobalObject, reason, functionObject, callData, thisValue, args, returnedException);
+            }
```

Comment:
This seems very heavy weight to have to do this check and use an RAII scope every time.  Is there no better location where we can set and clear the flag in the Document's VM on page load and reload (so that it's done only once)?